src/store/proposals/sagas/addProposal.ts

```diff
@@ -137,15 +143,15 @@ export function* checkAddEligibility() {
 }
 
 function* triggerFetch() {
-  yield* put(proposalsActions.checkAddProposal());
+  yield* put(proposalsActions.checkEligibility());
 }
 
 function* triggerUpdate() {
-  yield* put(proposalsActions.checkAddProposal());
+  yield* put(proposalsActions.checkEligibility());
 }
 
-export const watchAddProposals = createWatcherSaga({
+export const watchEligibility = createWatcherSaga({
   fetchSaga: triggerFetch,
   updateSaga: triggerUpdate,
```

Comment:
You can remove the `triggerUpdate` saga and use the `triggerFetch` in both places
```suggestion
  updateSaga: triggerFetch,
```